papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/service/SprintsRestService.java

```diff
@@ -110,8 +105,56 @@
         return ResponseEntity.ok(sprintMapper.mapToApi(changedSprint));
     }
 
+    public ResponseEntity<SprintSummaryDto> getSprintSummary(Long projectId, Long sprintId) {
+        Sprint sprint = getValidSprint(sprintId);
+        Project project = projectsRestService.getValidProject(projectId);
+
+        if (project.isAdmin(userContext.getUser())) {
+            throw new ForbiddenAccessException();
+        }
+
+        List<UserAvailabilityDto> userAvailabilityDtos = availabilityRestService.getUserAvailabilities(sprintId);
+        Duration totalAvailableTime = sprintService.getTotalAvailableTimeBySprintId(sprintId);
+
+        //to be expanded in issue IO2019TEAM-100
+        double coefficient = sprint.getTimePlanned().dividedBy(sprint.getTimeBurned());
+
+        return ResponseEntity.ok(new SprintSummaryDto().membersAvailability(userAvailabilityDtos)
+                .totalAvailableTime(totalAvailableTime.toMinutes())
+                .sprintCoefficient(coefficient)
+        );
+    }
+
+    public ResponseEntity<SprintDto> modifySprint(SprintDto sprintDto, Long projectId, Long sprintId) {
+        Project project = projectsRestService.getValidProject(projectId);
+
+        if (!project.isAdmin(userContext.getUser())) {
+            throw new ForbiddenAccessException();
+        }
+
+        if (sprintDto.getSprintState() == SprintStateDto.CLOSED) {
+            Duration timeBurned = Duration.ofMinutes(
+                    AssertionUtil.requireNonNegative("timeBurned", sprintDto.getTimeBurned())
+            );
+
+            @SuppressWarnings({"MultipleStringLiterals"})
+            Duration timePlanned = Duration.ofMinutes(
+                    AssertionUtil.requireNonNegative("timePlanned", sprintDto.getTimePlanned())
+            );
+
+            return closeSprint(sprintId, timeBurned, timePlanned);
+        }
+
+        throw new IllegalArgumentException("Invalid patch");
+    }
+
     private Sprint getValidSprint(Long sprintId) {
         return sprintService.getById(sprintId)
                 .orElseThrow(ResourceNotFoundException::new);
     }
+
+    public boolean isPreviousSprintOpen(Project project, LocalDateTimePeriod durationPeriod) {
```

Comment:
To sprawdza co innego. `noOverlapWithExistingSprints` czy coś?

Reply:
Poprawione